temporalio/converter.py

```diff
@@ -776,3 +801,218 @@ def _type_hints_from_func(
             return (None, ret)
         args.append(arg_hint)
     return args, ret
+
+
+def value_to_type(hint: Type, value: Any) -> Any:
+    """Convert a given value to the given type hint.
+
+    This is used internally to convert a raw JSON loaded value to a specific
+    type hint.
```

Comment:
Isn't this more like?
```suggestion
    This is used internally to convert a raw JSON loaded value to an instance
    of a specific type.
```

Reply:
```suggestion
    This is used internally to convert a raw JSON loaded value using a specific
    type hint.
```

That may be clearer. Will change if there is another commit here.